src/nbl/video/CVulkanPhysicalDevice.h

```diff
@@ -101,72 +100,163 @@ class CVulkanPhysicalDevice final : public IPhysicalDevice
         }
     }
             
-    inline VkPhysicalDevice getInternalObject() const { return m_vkphysdev; }
+    inline VkPhysicalDevice getInternalObject() const { return m_vkPhysicalDevice; }
             
     E_API_TYPE getAPIType() const override { return EAT_VULKAN; }
 
-    std::vector<ISurface::SFormat> getAvailableFormatsForSurface(const ISurface* surface) const override
+    // Todo(achal)
+    IDebugCallback* getDebugCallback() override { return nullptr; }
```

Comment:
if your debug messenger is part of the Instance (API connection) then your debug callback will live there, and you need to get it back from the API connection reference

This necessitates a .cpp file (because now you need to access methods of the "logical parent")